src/main/java/ch/njol/skript/doc/JSONGenerator.java

```diff
@@ -0,0 +1,247 @@
+package ch.njol.skript.doc;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.lang.SkriptEventInfo;
+import ch.njol.skript.lang.SyntaxElement;
+import ch.njol.skript.lang.SyntaxElementInfo;
+import ch.njol.skript.lang.function.Functions;
+import ch.njol.skript.lang.function.JavaFunction;
+import ch.njol.skript.registrations.Classes;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import org.eclipse.jdt.annotation.Nullable;
+import org.skriptlang.skript.lang.structure.Structure;
+import org.skriptlang.skript.lang.structure.StructureInfo;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * Generates JSON docs
+ */
+public class JSONGenerator extends DocumentationGenerator {
+
+	public JSONGenerator(File templateDir, File outputDir) {
+		super(templateDir, outputDir);
+	}
+
+	/**
+	 * Coverts a String array to a JsonArray
+	 * @param strings the String array to convert
+	 * @return the JsonArray containing the Strings
+	 */
+	private static @Nullable JsonArray convertToJsonArray(String @Nullable[] strings) {
```

Comment:
```suggestion
	private static @Nullable JsonArray convertToJsonArray(String @Nullable [] strings) {
```
Could use a contract